common/flatpak-dir.c

```diff
@@ -3936,18 +3922,50 @@ _flatpak_dir_find_new_flatpakrepos (FlatpakDir *self, OstreeRepo *repo)
         {
           g_autofree char *remote_name = g_strndup (name, strlen (name) - strlen (SYSCONF_REMOTES_FILE_EXT));
 
-          if (remotes && g_strv_contains ((const char * const *)remotes, remote_name))
-            continue;
-
-          if (applied_remotes && g_strv_contains ((const char * const *)applied_remotes, remote_name))
+          if (remotes && g_strv_contains (remotes, remote_name))
             continue;
 
-          if (flatpakrepos == NULL)
-            flatpakrepos = g_hash_table_new_full (g_str_hash, g_str_equal, g_free, g_object_unref);
+          if (*flatpakrepos == NULL)
+            *flatpakrepos = g_hash_table_new_full (g_str_hash, g_str_equal, g_free, g_object_unref);
 
-          g_hash_table_insert (flatpakrepos, g_steal_pointer (&remote_name), g_file_enumerator_get_child (dir_enum, file_info));
+          g_hash_table_insert (*flatpakrepos, g_steal_pointer (&remote_name), g_file_enumerator_get_child (dir_enum, file_info));
         }
     }
+}
+
+static GHashTable *
+_flatpak_dir_find_new_flatpakrepos (FlatpakDir *self, OstreeRepo *repo)
+{
+  g_autoptr(GHashTable) flatpakrepos = NULL;
+  g_autofree char *conf_dir_str = NULL;
+  g_auto(GStrv) ostree_remotes = NULL;
+  g_auto(GStrv) applied_remotes = NULL;
+  g_autoptr(GPtrArray) remotes = NULL;
+
+  g_assert (repo != NULL);
+
+  /* Predefined remotes only applies for the default system installation */
+  if (self->user ||
+      g_strcmp0 (self->extra_data->id, SYSTEM_DIR_DEFAULT_ID) != 0)
```

Comment:
This is the semantic change that causes trouble. `self->extra_data` can be `NULL` in which case this crashes. The `g_strcmp0` also guards against the id of the extra_data being NULL.

```patch
diff --git ./common/flatpak-dir.c ../common/flatpak-dir.c
index 4400a2a2..b9cc5a7c 100644
--- ./common/flatpak-dir.c
+++ ../common/flatpak-dir.c
@@ -3947,7 +3947,8 @@ _flatpak_dir_find_new_flatpakrepos (FlatpakDir *self, OstreeRepo *repo)
 
   /* Predefined remotes only applies for the default system installation */
   if (self->user ||
-      g_strcmp0 (self->extra_data->id, SYSTEM_DIR_DEFAULT_ID) != 0)
+      (self->extra_data &&
+       g_strcmp0 (self->extra_data->id, SYSTEM_DIR_DEFAULT_ID) != 0))
     return NULL;
 
   ostree_remotes = ostree_repo_remote_list (repo, NULL);
```

Reply:
Uf, stupid refactor mistake. Applied!